### PR TITLE
print gen_snapshot exit code; fix a missing type annotation

### PR DESCRIPTION
**Pull request by yjbanov**
/cc @devoncarew

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -260,7 +260,7 @@ class IOSDevice extends Device {
           forwardDiagPort = _acquireAndForwardPort(ProtocolDiscovery.kDiagnosticService,
                                                    debuggingOptions.diagnosticPort);
         } else {
-          forwardDiagPort = new Future.value(null);
+          forwardDiagPort = new Future<int>.value(null);
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/ios/devices.dart]: Does `flutter analyze` not catch this?

Comment (yjbanov) [inline packages/flutter_tools/lib/src/ios/devices.dart]: It does now. An older version of the analyzer had a bug and failed to catch this. We started seeing failures after we picked up a new analyzer.

**Pull request merged**